### ISSUE TITLE
Using with postversion: option to amend the previous commit?

### ISSUE THREAD

**Issue by sanjaypojo** [axelpale/genversion#7]
Hi, thanks for the package, it works great. Is there an option to automatically amend the version commit if it's used as a postversion script? In `package.json` we add `postversion: genversion` so that `genversion` is run whenever the `npm version` command is run.

Ex: Suppose we now run `npm version minor` from the command line. It updates the version number in `package.json`. Then genversion is run as it is the postversion script. It updates a file in `src` with this new version number. However, this change is not commited. Ideally, it would be added to the version commit. Can the script have an `-amend` option that does this automatically? It should just amend the previous commit with the change. Other similar packages do this, for example, [`react-native-version`](https://github.com/stovmascript/react-native-version)

**Comment by axelpale** [comment]
Nice feature request! I am unable to look into this in the few upcoming weeks but maybe after it... If you have an idea how to implement it, a PR or some sketching are always welcome. Maybe `--amend` with two dashes?

**Comment by sanjaypojo** [comment]
Thanks for the quick response @axelpale. Unfortunately, I don't have time at the moment for a full blown PR with tests, but here's a quick sketch of the approach, it should be pretty straightforward. After all changes have been made:

```javascript
if ("--amend" flag exists) {
  execSync(`git add ${pathToGenVersionFile}`)
  execSync("git commit --amend --no-edit")
}
```

**Comment by sanjaypojo** [comment]
@axelpale actually there's a little more to it if we want to also preserve the git tag created by `npm version`. We could take inspiration from react-native-version:

https://github.com/stovmascript/react-native-version/blob/9e72dd72a4a65caeb0237711fa5bf6cff7e1953f/index.js#L584

**Comment by sanjaypojo** [comment]
Here's a fully working script.

```sh
#!/bin/sh

# Get git tag created by version commit
lastTag=$(git describe --exact-match --abbrev=0)
# Run genversion to update version number
genversion app/lib/version.js
# Amend version commit
git add app/lib/version.js
git commit --amend --no-edit
# Amend git tag produced by version commit
git tag -fa $lastTag -m $lastTag
```

To use, simply add this shell script to a file called `postversion.sh` and then add it to `package.json`
```json
{
  ...
  "scripts": {
    "postversion": "./postversion.sh"
    ...
  }
  ...
}
```

Then, when you run `npm version minor`, genversion is automatically called and correctly amends the commit message as well.

PS: once you create `postversion.sh`, ensure it's executable by running `chmod +x postversion.sh`

**Comment by axelpale** [comment]
Thank you for all the work! New docs released in v3.2.0
